src/API/Site/Controllers/CountryCodeTrait.php

```diff
@@ -35,24 +36,29 @@ protected function validate_country_code( string $country ): void {
 	}
 
 	/**
-	 * Validate that a country or a list of countries is valid and supported.
+	 * Validate that a country or a list of countries is valid and supported,
+	 * and also validate the data by the built-in validation of WP REST API with parameter’s schema.
 	 *
-	 * @param mixed $countries                An individual string or an array of strings.
-	 * @param bool  $check_supported_country  Whether to check the country is supported.
+	 * @param bool    $check_supported_country  Whether to check the country is supported.
+	 * @param mixed   $countries                An individual string or an array of strings.
+	 * @param Request $request                  The request to validate.
+	 * @param string  $param                    The parameter name, used in error messages.
 	 *
 	 * @return mixed
 	 * @throws Exception            When the country is not supported.
 	 * @throws OutOfBoundsException When the country code cannot be found.
 	 */
-	protected function validate_country_codes( $countries, bool $check_supported_country ) {
+	protected function validate_country_codes( bool $check_supported_country, $countries, $request, $param ) {
+		$valid = rest_validate_request_arg( $countries, $request, $param );
+
+		if ( true !== $valid ) {
+			return $valid;
+		}
```

Comment:
I'd suggest renaming this as `$validation_result`, `$validation` or something like that. Since it can return a WP_Error object, hence is not "valid".

Reply:
Renamed in https://github.com/woocommerce/google-listings-and-ads/pull/2003/commits/bdfb3651bf14b1066336b92839e2b4b10eff8498.